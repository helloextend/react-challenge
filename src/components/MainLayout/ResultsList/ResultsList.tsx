import React, { FC } from 'react'
import styled from '@emotion/styled'
import Heart from '../../Heart';
import { getRandomAmtFromArray } from '../../../assets/custom';

interface Props {
  imgUrls: Array<string>,
  clickHeart: (imgUrl: string) => void,
  favorites?: Array<string>,
  columnCount: number
}

const ResultsList: FC<Props> = ({ imgUrls, clickHeart, favorites, columnCount }) => {
  const renderImages = (urlsArray) => {
    let images = []
    for (let i = 0; i < 10; i++) {
      if (imgUrls[i]) {
        const iconColor = favorites.indexOf(imgUrls[i]) === -1 ? "whiteHeartIcon" : "redHeartIcon"

        images.push(
          <ImgContainer key={i}>
            <DogImg src={urlsArray[i]} alt="dog" />
            <Heart onClickHeart={clickHeart} url={urlsArray[i]} style={heartStyle} icon={iconColor} alt="white heart icon" />
          </ImgContainer>
        )
      }
    }
    return images;
  }

  const DogList = styled.div({
    columnCount: columnCount,
    columnGap: '1em',
    margin: '36px 0'
  })

  return (
    <DogList>
      {imgUrls[0] && renderImages(imgUrls)}
    </DogList>
  )
}





const DogImg = styled.img({
  width: '100%',
  height: 'auto',
  margin: '0 0 1em',
})

const ImgContainer = styled.div({
  position: 'relative',
})

const heartStyle = {
  position: 'absolute',
  bottom: '34px',
  right: '10px',
}




export default ResultsList
