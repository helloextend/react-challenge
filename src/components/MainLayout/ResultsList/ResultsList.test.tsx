import * as React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import ResultsList from './ResultsList'


const baseProps = {
  favorites: [
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_1003.jpg',
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_1007.jpg',
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_1023.jpg',
  ],
  clickHeart: jest.fn(),
  imgUrls: [
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_10822.jpg',
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_10832.jpg',
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_10982.jpg',
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_11006.jpg',
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_11172.jpg',
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_11182.jpg',
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_1126.jpg',
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_1128.jpg',
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_11432.jpg',
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_1145.jpg',
  ],
  columnCount: 3
};



describe('ResultsList component', () => {
  it('renders without crashing', () => {
    shallow(<ResultsList {...baseProps} />)
  });

  it('renders the UI as expected when given no urls', () => {
   const tree = renderer
      .create(<ResultsList {...baseProps} imgUrls={[]}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders 10 dog images when given an array of 10 urls', () => {
    const wrapper = shallow(<ResultsList {...baseProps} />)
    expect(wrapper.find('Styled(img)')).toHaveLength(10)
  })

})