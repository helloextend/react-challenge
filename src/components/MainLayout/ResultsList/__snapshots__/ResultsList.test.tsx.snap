// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList component renders the UI as expected when given no urls 1`] = `
<div
  className="css-r0qpvp"
/>
`;
